code/game/objects/items/weapons/melee/misc.dm

```diff
@@ -125,3 +125,31 @@
 
 	playsound(src.loc, 'sound/weapons/batonextend.ogg', 50, 1)
 	add_fingerprint(user)
+
+/obj/item/weapon/melee/supermatter_sword
+	name = "supermatter sword"
+	desc = "In a station full of bad ideas, this might just be the worst."
+	icon = 'icons/obj/weapons.dmi'
+	icon_state = "telebaton_0" //Needed!
+	item_state = null //Needed!
+	slot_flags = null
+	w_class = 4
+	var/obj/machinery/power/supermatter_shard/shard
+
+
+/obj/item/weapon/melee/supermatter_sword/New()
+	..()
+	shard = new /obj/machinery/power/supermatter_shard(src)
+
+/obj/item/weapon/melee/supermatter_sword/afterattack(target)
+	..()
+	shard.Bumped(target)
+
+/obj/item/weapon/melee/supermatter_sword/throw_impact(target)
+	..()
+	shard.Bumped(target)

```

Comment:
Even if you huck it clear cross the screen you can't escape the radiation, which has a radius of 10 tiles.